src/pycram/knowledge/knowledge_engine.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import annotations
+
+import inspect
+
+import rospy
+from anytree import PreOrderIter
+from typeguard import check_type, TypeCheckError
+
+from ..datastructures.partial_designator import PartialDesignator
+from ..datastructures.property import Property, ResolvedProperty, PropertyOperator
+from .knowledge_source import KnowledgeSource
+from typing_extensions import Type, Callable, List, TYPE_CHECKING, Dict, Any
+
+from ..failures import KnowledgeNotAvailable, ReasoningError
+# This import is needed since the subclasses of KnowledgeSource need to be imported to be known at runtime
+from .knowledge_sources import *
```

Comment:
if stuff like this is needed i recommend to import it in the __init__.py I got the same problems for the subclass json serializer but the best way to resolve the conflict for me was  to import what i can in the init of a module and the rest (cyclic dependency stuff) has to be done by the user when he wants to use that class.